libs/dh/market-participant/actors/feature-actors/src/lib/edit/dh-actors-edit-actor-modal.component.ts

```diff
@@ -14,25 +14,142 @@
+import { tap } from 'rxjs';
+import { RxLet } from '@rx-angular/template/let';
+import { WattToastService } from '@energinet-datahub/watt/toast';
+import { WattTextFieldComponent } from '@energinet-datahub/watt/text-field';
+import { WattFieldErrorComponent } from '@energinet-datahub/watt/field';
 
 @Component({
   standalone: true,
   selector: 'dh-actors-edit-actor-modal',
   templateUrl: './dh-actors-edit-actor-modal.component.html',
-  styles: [``],
-  imports: [WATT_MODAL, CommonModule],
+  providers: [DhMarketParticipantActorsEditActorDataAccessApiStore],
+  styles: [
+    `
+      .actor-field {
+        width: 80%;
+      }
+
+      .contact-field {
+        width: 60%;
+      }
+
+      .phone-field {
+        width: 35%;
+      }
+
+      .email-suffix {
+        padding-right: var(--watt-space-s);
+      }
+    `,
+  ],
+  imports: [
+    WATT_MODAL,
+    ReactiveFormsModule,
+    WattButtonComponent,
+    WattTextFieldComponent,
+    WattFieldErrorComponent,
+    RxLet,
+    CommonModule,
+    TranslocoModule,
+  ],
 })
 export class DhActorsEditActorModalComponent {
+  private readonly dataAccessStore = inject(DhMarketParticipantActorsEditActorDataAccessApiStore);
+  private readonly formBuilder = inject(FormBuilder);
+  private readonly transloco = inject(TranslocoService);
+  private readonly toastService = inject(WattToastService);
+
   @ViewChild(WattModalComponent)
   innerModal: WattModalComponent | undefined;
 
   @Input() actor: DhActorExtended | undefined;
 
-  open(): void {
+  actorForm = this.formBuilder.group({
+    name: ['', Validators.required],
+    departmentName: ['', Validators.required],
+    departmentEmail: ['', [Validators.required, Validators.pattern('^[a-zA-Z0-9._%+-]+')]],
+    departmentPhone: ['', [Validators.required]],
+  });
+
+  isLoading = true;
+  emailDomain = '';
+
+  formInitialValues$ = this.dataAccessStore.actorEditableFields$.pipe(
+    tap((queryResult) => {
+      this.isLoading = queryResult.loading;
+
+      if (!queryResult.data || queryResult.loading) return;
+
+      this.emailDomain = queryResult.data.actorById.organization.domain;
+      this.actorForm.markAsUntouched();
+      this.actorForm.patchValue({
+        name: queryResult.data.actorById.name,
+        departmentName: queryResult.data.actorById.contact?.name,
+        departmentPhone: queryResult.data.actorById.contact?.phone,
+        departmentEmail: queryResult.data.actorById.contact?.email.replace(
+          `@${queryResult.data.actorById.organization.domain}`,
+          ''
+        ),
+      });
+    })
+  );
+
+  open() {
+    if (!this.actor) return;
+    this.dataAccessStore.load(this.actor.id);
     this.innerModal?.open();
   }
+
+  save() {
+    if (
+      !this.actor ||
+      !this.actorForm.value.name ||
+      !this.actorForm.value.departmentName ||
+      !this.actorForm.value.departmentPhone ||
+      !this.actorForm.value.departmentEmail
+    )
+      return;
```

Comment:
suggestion: can also be used for early return,

```
if (this.actorForm.invalid) {
  return;
}
```

Reply:
Cannot, typesystem cannot see that the fields are not null.